app/src/main/java/de/westnordost/streetcomplete/quests/diet_type/AddVegan.kt

```diff
@@ -10,8 +10,12 @@ class AddVegan : OsmFilterQuestType<DietAvailability>() {
     override val elementFilter = """
         nodes, ways with
         (
-          amenity ~ restaurant|cafe|fast_food|pub and diet:vegetarian ~ yes|only
-          or amenity = ice_cream
+          amenity = ice_cream
+          or diet:vegetarian ~ yes|only and
+          (
+            amenity ~ restaurant|cafe|fast_food
+            or amenity = pub and food != no
```

Comment:
should be `food=yes`, not `food != no`. Otherwise this is also asked if `food` is missing.

Reply:
Would `food and food != no` work? There are currently 348 values which aren't no but appear to be positive/true.

Reply:
but are you sure that those values are/would be used on `amenity=pub`? Is maybe the `food` key also used in another context?

Reply:
Pizza, which looked like the most popular likely value, has four hits, when I do a query; restaurant has 7. So aren't we better off including them too? Even if they weren't used on pub now, presumably they'd show up in iD and whatever as suggestions when you used the food tag, so they might get added if they weren't present already.